### PR TITLE
Add goes specific metadata

### PR DESCRIPTION
**Pull request by gadomski**
These are namespaced with `goes:`.

### CODE REVIEW

**Comment by lossyrob** [comment]
These are interested in these facts about a scene to include:
- What [scan mode](https://www.goes-r.gov/users/abiScanModeInfo.html) the satellite is in (e.g. Mode 4, continuous full disk, or Mode 6, "10 minute flex mode")
- Image Type (Full Disk, CONUS, Mesos)
- Mesoscale Image Number
- Cyclone Identifier Number

These can be pulled from the filename of the netcdf. Refer to [this document](https://www.goes-r.gov/products/docs/PUG-L2+-vol5.pdf), page 609 by the bottom numbers (646 of the pdf), "A.1 Level 2+ Product Filenames".

For enumerations (like image type, scan mode etc) it would be good to include Collection summaries for those properties. This stactools package doesn't create a Collection, but using Enums for these property will make it clear how to list them in the Collection derive elsewhere.

**Comment by gadomski** [comment]
> These are interested in these facts about a scene to include:

>    What scan mode the satellite is in (e.g. Mode 4, continuous full disk, or Mode 6, "10 minute flex mode")
>   Image Type (Full Disk, CONUS, Mesos)
>    Mesoscale Image Number
>    Cyclone Identifier Number

> These can be pulled from the filename of the netcdf. Refer to this document, page 609 by the bottom numbers (646 of the pdf), "A.1 Level 2+ Product Filenames".

These are all included (see https://github.com/stactools-packages/goes/pull/12/files#diff-fcd8900fc8d14bab79e38362aa07a61c1bf1ce3cea05ccd0aab72662fbcf9f1dR37-R50 for the metadata fields) with the exception of Cyclone Identifier Number, which is only valid for the HIE product, so I didn't include it.

> For enumerations (like image type, scan mode etc) it would be good to include Collection summaries for those properties. This stactools package doesn't create a Collection, but using Enums for these property will make it clear how to list them in the Collection derive elsewhere.

Ok, will add.

**Comment by lossyrob** [comment]
From the tests it seems like `timeline-id` is the scan mode, and `scene-id` is the image type? Are these the official names or should they be named more clearly with the top level doc names?

**Comment by gadomski** [comment]
I took the names from the variables in the NetCDF file, so they're "official" in that sense -- the variable names and descriptions are documented in the data fields for each L2+ product, e.g.: Screen Shot 2021-09-07 at 8 44 27 AM

File: src/stactools/goes/enums.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class ImageType(Enum):
+    FULL_DISK = enum.auto()
+    CONUS = enum.auto()
+    MESOSCALE = enum.auto()
+
+
+class Mode(Enum):
+    ABI_MODE_3 = 3
+    ABI_MODE_4 = 4
+    ABI_MODE_6 = 6
```
Comment (lossyrob) [inline src/stactools/goes/enums.py:14]: I think these should be string Enums and set into the properties with string values parsed from the possible values. E.g. from the tests it looks like "ABI Mode 6" etc for mode and `Mesoscale` etc for the ImageType

I believe `goes:mesoscale-image-number` is also a restricted value, so should use an enum?

**Pull request merged**